libnd4j/include/ops/declarable/generic/transforms/layer_norm.cpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2019 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+//
+// @author Paul Dubs
+//
+
+#include <op_boilerplate.h>
+#if NOT_EXCLUDED(OP_layer_norm)
+
+#include <ops/declarable/CustomOperations.h>
+#include <ops/declarable/helpers/reverse.h>
+
+
+namespace nd4j {
+namespace ops  {
+
+    CONFIGURABLE_OP_IMPL(layer_norm, 2, 1, false, 0, -1) {
```

Comment:
No shape function for layer_norm forward pass?

Reply:
CONFIGURABLE_OP_IMPL creates it automatically